TCAT/Views/RouteSelectionView.swift

```diff
@@ -156,139 +122,80 @@ class RouteSelectionView: UIView {
-
-        positionSwapButton(usingFromSearchBar: fromSearchbar, usingLine: line)
-    }
-
-    private func positionLabelHorizontally(_ label: UILabel) {
-        let oldFrame = label.frame
-        let newFrame = CGRect(x: leadingSpace, y: oldFrame.minY, width: oldFrame.width, height: oldFrame.height)
-
-        label.frame = newFrame
-    }
-
-    private func postionSolidCircleHorizontally(usingFromLabel fromLabel: UILabel) {
-        solidCircle.center.x = fromLabel.frame.maxX + solidCircleLeftSpace + (solidCircle.frame.width/2)
-    }
-
-    private func positionFromSearchbar(usingSolidCircle solidCircle: Circle) {
-        let oldFrame = fromSearchbar.frame
-        let newFrame = CGRect(x: solidCircle.frame.maxX + solidCircleRightSpace, y: topSpace, width: oldFrame.width, height: oldFrame.height)
-
-        fromSearchbar.frame = newFrame
-    }
-
-    private func positionToSearchbar(usingFromSearchbar fromSearchbar: UIButton) {
-        let oldFrame = toSearchbar.frame
-        let newFrame = CGRect(x: fromSearchbar.frame.minX, y: fromSearchbar.frame.maxY + leadingSpace, width: oldFrame.width, height: oldFrame.height)
-
-         toSearchbar.frame = newFrame
-    }
-
-    private func positionSolidCircleVertically(usingFromSearchbar fromSearchbar: UIButton) {
-        solidCircle.center.y = fromSearchbar.center.y
-    }
-
-    private func positionBorderedCircle(usingSolidCircle solidCircle: Circle, usingToSearchbar toSearchbar: UIButton) {
-        borderedCircle.center.x = solidCircle.center.x
-        borderedCircle.center.y = toSearchbar.center.y
-    }
-
-    private func positionLine(usingSolidCircle solidCircle: Circle, usingBorderedCircle borderedCircle: Circle) {
-        line.center.x = solidCircle.center.x
-
-        let oldFrame = line.frame
-        let newFrame = CGRect(x: oldFrame.minX, y: solidCircle.center.y, width: oldFrame.width, height: 1 + borderedCircle.frame.minY - solidCircle.center.y)
-
-        line.frame = newFrame
-    }
-
-    private func positionLabelVertically(_ label: UILabel, usingSearchbar searchbar: UIButton) {
-        label.center.y = searchbar.center.y
-    }
-
-    private func positionSearchbarView(usingFromSearchbar fromSearchbar: UIButton, usingToSearchbar toSearchbar: UIButton) {
-        let oldFrame = searcbarView.frame
-        let newFrame = CGRect(x: 0, y: lineWidth, width: oldFrame.width, height: topSpace + fromSearchbar.frame.height + leadingSpace + toSearchbar.frame.height + (3*topSpace/4))
-
-        searcbarView.frame = newFrame
-    }
+    private func setupConstraints() {
+        let separatorHeight: CGFloat = 1.0
 
-    private func positionDatepickerButton(usingSearchbarView searchbarView: UIView) {
-        let oldFrame = datepickerButton.frame
-        let newFrame = CGRect(x: 0, y: searchbarView.frame.maxY + lineWidth, width: oldFrame.width, height: oldFrame.height)
+        swapButton.snp.makeConstraints { make in
+            make.size.equalTo(CGSize(width: 20, height: 25))
```

Comment:
can you move all these numbers into constants at the top of the function just like line 126